MdePkg/Test/Mock/Include/GoogleTest/Protocol/MockPciIo.h

```diff
@@ -0,0 +1,241 @@
+    )
+    );
+
+  MOCK_FUNCTION_DECLARATION (
+    EFI_STATUS,
+    Unmap,
+    (
+     IN EFI_PCI_IO_PROTOCOL           *This,
+     IN  VOID                         *Mapping
+    )
+    );
+
+  MOCK_FUNCTION_DECLARATION (
+    EFI_STATUS,
+    AllocateBuffer,
+    (
+     IN EFI_PCI_IO_PROTOCOL           *This,
+     IN  EFI_ALLOCATE_TYPE            Type,
+     IN  EFI_MEMORY_TYPE              MemoryType,
+     IN  UINTN                        Pages,
+     OUT VOID                         **HostAddress,
+     IN  UINT64                       Attributes
+    )
+    );
+
+  MOCK_FUNCTION_DECLARATION (
+    EFI_STATUS,
+    FreeBuffer,
+    (
+     IN EFI_PCI_IO_PROTOCOL           *This,
+     IN  UINTN                        Pages,
+     IN  VOID                         *HostAddress
+    )
+    );
+
+  MOCK_FUNCTION_DECLARATION (
+    EFI_STATUS,
+    Flush,
+    (
+     IN EFI_PCI_IO_PROTOCOL  *This
+    )
+    );
+
+  MOCK_FUNCTION_DECLARATION (
+    EFI_STATUS,
+    GetLocation,
+    (
+     IN EFI_PCI_IO_PROTOCOL          *This,
+     OUT UINTN                       *SegmentNumber,
+     OUT UINTN                       *BusNumber,
+     OUT UINTN                       *DeviceNumber,
+     OUT UINTN                       *FunctionNumber
+    )
+    );
+
+  MOCK_FUNCTION_DECLARATION (
+    EFI_STATUS,
+    Attributes,
+    (
+     IN EFI_PCI_IO_PROTOCOL                       *This,
+     IN  EFI_PCI_IO_PROTOCOL_ATTRIBUTE_OPERATION  Operation,
+     IN  UINT64                                   Attributes,
+     OUT UINT64                                   *Result OPTIONAL
+    )
+    );
+
+  MOCK_FUNCTION_DECLARATION (
+    EFI_STATUS,
+    GetBarAttributes,
+    (
+     IN EFI_PCI_IO_PROTOCOL             *This,
+     IN  UINT8                          BarIndex,
+     OUT UINT64                         *Supports  OPTIONAL,
+     OUT VOID                           **Resources OPTIONAL
+    )
+    );
+
+  MOCK_FUNCTION_DECLARATION (
+    EFI_STATUS,
+    SetBarAttributes,
+    (
+     IN EFI_PCI_IO_PROTOCOL              *This,
+     IN     UINT64                       Attributes,
+     IN     UINT8                        BarIndex,
+     IN OUT UINT64                       *Offset,
+     IN OUT UINT64                       *Length
+    )
+    );
+};
+
+MOCK_INTERFACE_DEFINITION (MockEfiPciIoProtocolAccess);
+MOCK_FUNCTION_DEFINITION (MockEfiPciIoProtocolAccess, PciIoProtocolRead, 6, EFIAPI);
+MOCK_FUNCTION_DEFINITION (MockEfiPciIoProtocolAccess, PciIoProtocolWrite, 6, EFIAPI);
+
+MOCK_INTERFACE_DEFINITION (MockEfiPciIoProtocolConfigAccess);
+MOCK_FUNCTION_DEFINITION (MockEfiPciIoProtocolConfigAccess, PciIoRead, 5, EFIAPI);
+MOCK_FUNCTION_DEFINITION (MockEfiPciIoProtocolConfigAccess, PciIoWrite, 5, EFIAPI);
+
+MOCK_INTERFACE_DEFINITION (MockEfiPciIoProtocol);
+MOCK_FUNCTION_DEFINITION (MockEfiPciIoProtocol, PollIoMem, 8, EFIAPI); // todo I can just reuse this?
+MOCK_FUNCTION_DEFINITION (MockEfiPciIoProtocol, MockCopyMem, 7, EFIAPI);

```

Comment:
Renaming a service in mocked protocol has side effects on unit test code to know that the service name is not the same.  Since this should be part of a C++ class, why is there a name collision with C CopyMem()?

Reply:
Can you provide some examples of the EXPECT_CALL() usages with Protocol/PPI/ServiceTable?

Reply:
We've also done things like prepending the function names with "g" and the library initials like for BootServices (Example: gBS_LocateProtocol).
https://github.com/tianocore/edk2/blob/master/NetworkPkg/UefiPxeBcDxe/GoogleTest/PxeBcDhcp6GoogleTest.cpp#L547

But I agree this can be standardized.